### ISSUE TITLE
Show all labels in a PieChart

### ISSUE THREAD

**Issue by ChandraTheDev** [beto-rodriguez/Live-Charts#296]
I tried to print the Piechart as converting it to Image control. the Output is like below.

![capture](https://cloud.githubusercontent.com/assets/17137255/17462728/33511080-5cbf-11e6-80dd-65697f89642a.PNG)

But I need to know the percentage of each item. This can be known by mouseover in application. but while printing how can i get this?

my code is below:

 public BlockUIContainer GetBulkContainer(Visual target, double dpiX, double dpiY)
        {
            if (target == null)
            {
                return null;
            }

```
        if (target.GetType() == typeof(TextBlock))
        {
            TextBlock tb = new TextBlock();
            tb.Text = ((TextBlock)target).Text;
            return new BlockUIContainer(tb);
        }
        else
        {
            // render control content
            Rect bounds = VisualTreeHelper.GetDescendantBounds(target);
            RenderTargetBitmap rtb = new RenderTargetBitmap((int)(bounds.Width * dpiX / 96.0),
                                                            (int)(bounds.Height * dpiY / 96.0),
                                                            dpiX,
                                                            dpiY,
                                                            PixelFormats.Pbgra32);
            DrawingVisual dv = new DrawingVisual();
            using (DrawingContext ctx = dv.RenderOpen())
            {
                VisualBrush vb = new VisualBrush(target);

                ctx.DrawRectangle(vb, null, new Rect(new System.Windows.Point(), bounds.Size));
            }
            rtb.Render(dv);


            PngBitmapEncoder png = new PngBitmapEncoder();
            png.Frames.Add(BitmapFrame.Create(rtb));
            MemoryStream stream = new MemoryStream();
            png.Save(stream);

            System.Windows.Controls.Image im = new System.Windows.Controls.Image();
            BitmapImage imageSource = new BitmapImage();
            imageSource.BeginInit();
            imageSource.StreamSource = stream;
            imageSource.CacheOption = BitmapCacheOption.OnLoad;
            imageSource.EndInit();

            im.Source = imageSource;
            return new BlockUIContainer(im);
        }
```

**Comment by beto-rodriguez** [comment]
Well you can use DataLabels showing the percentage

http://lvcharts.net/App/examples/v1/wpf/Pie%20Chart

Notice every chart point has many useful properties, one of them is Participation it holds what you need

http://lvcharts.net/App/documentation/beta/LiveCharts-ChartPoint

**Comment by Chandu-cuddle** [comment]
I tried as you suggested . please check the below screenshot:
![capture](https://cloud.githubusercontent.com/assets/17137255/17463064/142d385c-5cc7-11e6-9a44-449c4eb18d90.PNG)

Here i cannot see all item percentages.
and how to use Participation property in Piechart?

**Comment by beto-rodriguez** [comment]
Some of them are hidden by the default LabelPoint function, it hides the label when it is les than 5% if I remember well.

If you need to show them all, then override it

    myPieSeries { LabelPoint = point => point.Participation.ToString("P"); }

**Comment by Chandu-cuddle** [comment]
Thanks a lot. I almost get everything.
if it is possible like below, it would be very helpful to me.

![capture](https://cloud.githubusercontent.com/assets/17137255/17463257/be1486d8-5cca-11e6-9969-c378fb369760.PNG)

**Comment by beto-rodriguez** [comment]
Not supported yet, will keep this issue open to add it in a future

**Comment by NickSavin** [comment]
I have got the same issue.


This will be a good solution:
![image](https://cloud.githubusercontent.com/assets/20172765/18478950/ce08a1cc-79db-11e6-97b2-1c317dd01123.png)

**Comment by beto-rodriguez** [comment]
The referenced commit fixes the issue where the series algorithm was not passing the label when the slice was small, I don't see any logic reason why i did that.

Besides that fixed, from version 0.9.2 I improved the labels position in for pie series, this will help to solve this issue, also keep an eye in #447, I am sure a combination of this fix, plus that issue will be very useful in many cases.

Now we can customize the position of labels in a PieSeries, options are: OutsideSlice and InsideSlice, here is a brief snippet and the result:

![untitled](https://cloud.githubusercontent.com/assets/10853349/22630320/94ea053a-ebbd-11e6-9afe-63b1510586c6.png)

    <lvc:PieChart Margin="50">
            <lvc:PieChart.Resources>
                <Style TargetType="lvc:PieSeries">
                    <Setter Property="DataLabels" Value="True"></Setter>
                    <Setter Property="FontSize" Value="16"></Setter>
                    <Setter Property="Foreground" Value="#303030"></Setter>
                    <Setter Property="LabelPosition" Value="OutsideSlice"></Setter>
                    <Setter Property="StrokeThickness" Value="0"></Setter>
                </Style>
            </lvc:PieChart.Resources>
            <lvc:PieChart.Series>
                <lvc:PieSeries Values="5"/>
                <lvc:PieSeries Values="6"/>
                <lvc:PieSeries Values="8"/>
                <lvc:PieSeries Values="3"/>
                <lvc:PieSeries Values="4"/>
                <lvc:PieSeries Values="5"/>
                <lvc:PieSeries Values="9"/>
                <lvc:PieSeries Values="5"/>
                <lvc:PieSeries Values="6"/>
                <lvc:PieSeries Values="8"/>
                <lvc:PieSeries Values="3"/>
                <lvc:PieSeries Values="4"/>
                <lvc:PieSeries Values="5"/>
                <lvc:PieSeries Values="9"/>
                <lvc:PieSeries Values="5"/>
                <lvc:PieSeries Values="6"/>
                <lvc:PieSeries Values="8"/>
                <lvc:PieSeries Values="3"/>
                <lvc:PieSeries Values="4"/>
                <lvc:PieSeries Values="5"/>
                <lvc:PieSeries Values="9"/>
            </lvc:PieChart.Series>
        </lvc:PieChart>
